### ISSUE TITLE
GPU process high CPU usage with sandboxing enabled (Windows 11)

### ISSUE THREAD

**Issue by Girofox** [Alex313031/Mercury#185]
### Issue
When launching Mercury on Windows (AVX2, SSE version doesn't matter) i have a constant 20 percent cpu usage in Task Manager because of the GPU process. It seems that the GPU process is doing something and keeps one CPU core busy, this is visible in `about:performance`. This is happening even on a fresh Mercury installation without any profiles and without any extension.

The interesting thing is that this does not happen in a virtual machine, on my laptop without Nvidia GPU or on Linux. Without hardware acceleration in Mercury settings the issue still persists though.

![image](https://github.com/Alex313031/Mercury/assets/43116061/315f5027-8a7e-4c7d-bf74-fe8c51f9b9c5)

### Investigation
With **troubleshoot mode** enabled via "Hamburger menu -> Help -> Troubleshoot mode" the CPU usage is normal. The profiler option is not helpful but in a prior Mercury build there was a Decoding process with a high CPU usage visible.

### Workaround (potential unsafe!)
When I set **security.sandbox.gpu.level** to 0 the issue is "fixed". So the GPU process sandboxing is to blame. The issue doesn't appear in normal Firefox.

![image](https://github.com/Alex313031/Mercury/assets/43116061/0d61f83b-5f37-4ddc-badc-bf93491ca73e)

**Comment by SashaXser** [comment]
Try rename mercury.exe to firefox.exe

**Comment by Girofox** [comment]
Yes, this worked. Do you know why this happens when the file is mercury.exe?

**Comment by SashaXser** [comment]
> Yes, this worked. Do you know why this happens when the file is mercury.exe?

It's very simple, drivers optimize programs for the exe name, for example you can replace firefox.exe with chrome.exe and the load balancing will change too. 

In general, the optimization is configurable in the drivers. 

I also suggest you try changing thorium.exe to chrome.exe, this should work much better because the driver knows it's Google Chrome and not something else.

**Comment by goodusername123** [comment]
If you have a Nvidia GPU I would recommend downloading this: https://github.com/Orbmu2k/nvidiaProfileInspector
and adding "**mercury.exe**" to the "**Mozilla Firefox**" which will achieve the same effect as renaming the EXE.

Same thing applies to [Thorium](https://thorium.rocks/) but you add it to the "**Google Chrome**" profile instead.

**Comment by M0n7y5** [comment]
I tried that but there is an issue ...
![image](https://github.com/Alex313031/Mercury/assets/17201053/347299a6-0be2-46b5-bbbb-2d1f37807d72)

**Comment by M0n7y5** [comment]
@Alex313031 please change the exe name to something like mercury_browser.exe since mercury.exe is already used by gotham knights game and can cause issues with NVidia drivers

![image](https://github.com/Alex313031/Mercury/assets/17201053/2b17a2e2-ee5f-45a0-b2c9-b9b4a30cb614)

**Comment by Girofox** [comment]
> @Alex313031 please change the exe name to something like mercury_browser.exe since mercury.exe is already used by gotham knights game and can cause issues with NVidia drivers
> 
> ![image](https://private-user-images.githubusercontent.com/17201053/328162206-2b17a2e2-ee5f-45a0-b2c9-b9b4a30cb614.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwMDkwMjQsIm5iZiI6MTcxNTAwODcyNCwicGF0aCI6Ii8xNzIwMTA1My8zMjgxNjIyMDYtMmIxN2EyZTItZWU1Zi00NWEwLWIyYzktYjliNGEzMGNiNjE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA2VDE1MTg0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2OWQ2ZjVlZWM4OGViNjY2ZGRkNzcxYTM1NjgzY2ZiNjdhZmYwZDE4NGRhZjRmNjY1M2EzOTZlZDZmODFjMjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yS93KQFL2-wePUduMoW5eWZP8UGDFwqfrTexVTbFGWk)

This explains why this issue is not present with Intel or inside a virtual machine. Maybe manually adding an application profile helps inside Nvidia control panel. But this is just a workaround.

**Comment by M0n7y5** [comment]
> ... Maybe manually adding an application profile just for mercury.exe helps inside Nvidia control panel. But this is just a workaround.

You cant add it to nvidia profile since name mercury.exe is already taken.

**Comment by Girofox** [comment]
With Nvidia Profile Inspector you can remove mercury.exe from Gotham Knights and then add mercury.exe to Firefox profile. Unsure if these changes stay after Nvidia driver updates are installed. 

Gotham Knights shouldn't react badly after these changes (I don't play this game). The main issue is the GPU sandboxing in Mercury which relies on the exact NVIDIA Firefox profile even if you manually adjust the Nvidia profile exactly.

### Removing mercury.exe from "Gotham Knights" profile (Nvidia)

- Select mercury.exe from remove button
- hit **Apply changes**

![image](https://github.com/Alex313031/Mercury/assets/43116061/5bdd840d-400b-4c65-b180-ae95e63ec3b2)
### Adding mercury.exe to "Mozilla Firefox" profile (Nvidia)
- Select mercury.exe from add button
- hit **Apply changes**
![image](https://github.com/Alex313031/Mercury/assets/43116061/74a9940a-39ec-4d35-af3f-5dd7aa54ddbe)

**Comment by Girofox** [comment]
Update: Reinstalling driver resets these customizations to Gotham Knights and Mercury profile sadly.
